### PR TITLE
Add ability to check if UI state got dirty

### PR DESCRIPTION
**Pull request by Arthraim**
KNOW ISSUE: treat state as immutable by default, which can solve my problem but could be a bad design

### CODE REVIEW

File: src/ui.js [hunk 1]
```diff
@@ -299,6 +304,17 @@ export default function ui(key, opts = {}) {
           return this._cachedUIProps
         }
 
+        isUIDirty() {
+          // Might not be a good idea to use Immutable here
+          const defaultUIState = new Map(this.getDefaultUIState(opts.state));
+          const currentUIState = getUIState(this.context.store.getState()).get(this.key);
+          if (opts.dirty) {
```
Comment (dfguo) [inline src/ui.js]: maybe add a documentation for this option so people know how to use this.

File: src/ui.js [hunk 2]
```diff
@@ -299,6 +304,17 @@ export default function ui(key, opts = {}) {
...
+            return opts.dirty(defaultUIState, currentUIState);
+          } else {
+            return !defaultUIState.equals(currentUIState);
```
Comment (dfguo) [inline src/ui.js:313]: who implemented equals?

File: src/ui.js [hunk 3]
```diff
@@ -299,6 +304,17 @@ export default function ui(key, opts = {}) {
           return this._cachedUIProps
         }
 
+        isUIDirty() {
+          // Might not be a good idea to use Immutable here
+          const defaultUIState = new Map(this.getDefaultUIState(opts.state));
```
Comment (dfguo) [inline src/ui.js]: i actually think it's a good idea, at least in our applications

Comment (Arthraim) [reply src/ui.js]: Cool

File: src/ui.js (hunk 2 above)
Comment (Arthraim) [reply src/ui.js:313]: Immutable did

**Pull request merged**